### PR TITLE
feat: transactional CREATE/DROP

### PR DESCRIPTION
**Pull request by usamoi**

### CODE REVIEW

**Comment by VoVAllen** [comment]
Can we add some tests for this?

**Comment by usamoi** [comment]
> Can we add some tests for this?

I do not know how to test it.

**Comment by cutecutecat** [comment]
> > Can we add some tests for this?
> 
> I do not know how to test it.

Maybe we can start from what does this issue solved? Before and after?

**Comment by usamoi** [comment]
> > > Can we add some tests for this?
> > 
> > 
> > I do not know how to test it.
> 
> Maybe we can start from what does this issue solved? Before and after?

The remote index created by CREATE will be removed if the transaction is aborted.

**Comment by cutecutecat** [comment]
In previous discussion, if a user enter `ctrl+c` before SQL `CREATE INDEX` returned. At backend, `pgvecto.rs` will generate a `leaked` index which is never used and unmaintained forever.

This PR would trigger a index cleaner every time user send `ctrl+c`.

To test it, we might need a python script to send `abort` and check whether the index file is cleared soon later, which could be done later.

File: crates/service/src/index/mod.rs
Comment (cutecutecat) [inline crates/service/src/index/mod.rs:1]: As `flush` caller is removed, do we still need `IndexView.flush()` and `InstanceView.flush()`?

Comment (usamoi) [reply crates/service/src/index/mod.rs:1]: It's just moved to `Commit`.

**Review by VoVAllen — APPROVED** [review]

**Pull request merged**